clients/drcachesim/tools/func_view.h

```diff
@@ -111,12 +111,17 @@ class func_view_t : public analysis_tool_t {
         std::string last_trace_module_name;
     };
     struct traced_info_t {
-        int id = -1;
         std::set<std::string> names;
-        int num_args = 0;
+        int num_args = -1; // Illegal value to mark uninitialized info structs.
         bool noret = false;
     };
 
+    // Lookup the information for the shard's last_func_id and return its traced_info_t
+    // struct.  Sets shard.error if the ID is not found (but still returns a traced_info_t
+    // reference).
+    traced_info_t &
+    get_info_for_last_func_id(shard_data_t *shard);
+
     static bool
     cmp_func_stats(const std::pair<int, func_stats_t> &l,
                    const std::pair<int, func_stats_t> &r);
```

Comment:
(I was going to say let's add a unit test with an invalid funclist file as a sanity test -- but I don't see existing unit tests for this tool?  They're all end-to-end tests with live apps and real tool inputs and output with no mocks.  It's beyond the scope of this PR but if you feel ambitious and want to add unit tests that would be nice.  Looks like several of these tools have no unit tests: grrr.)

Reply:
Observation noted!